### PR TITLE
Breath interacts with body temperature

### PR DESCRIPTION
**Pull request by nightred**
## About The Pull Request

Your breath out now interacts with your body temperature.
Breath out will now be your body temperature with an offset as it never quite matches.
This has a minor effect on the station temp, but a 3x3 room of monkeys cannot even heat it up 1c over 10 min. and atmos is plumped to manage heat.

## Why It's Good For The Game

More atmo interactions.
The proc existed for this but was left empty.

## Changelog
:cl:
tweak: Your breath out now interacts with your body temperature.
/:cl:

### CODE REVIEW

File: code/__DEFINES/atmospherics.dm [hunk 1]
```diff
@@ -39,10 +39,13 @@
...
 /// liters in a cell
 #define CELL_VOLUME				2500
+
 /// liters in a normal breath
-#define BREATH_VOLUME			0.5
+#define BREATH_VOLUME				0.5
 /// Amount of air to take a from a tile
-#define BREATH_PERCENTAGE		(BREATH_VOLUME/CELL_VOLUME)
+#define BREATH_PERCENTAGE			(BREATH_VOLUME/CELL_VOLUME)
+// Your breath is a few degrees less then body temperatur
```
Comment (bobbahbrown) [inline code/__DEFINES/atmospherics.dm]:
```suggestion
// Your breath is a few degrees less then body temperature
```

File: code/modules/mob/living/carbon/life.dm [hunk 2]
```diff
@@ -302,7 +298,13 @@
 
 //Fourth and final link in a breath chain
 /mob/living/carbon/proc/handle_breath_temperature(datum/gas_mixture/breath)
-	return
+	// the air you breath out should match your body tempurature
```
Comment (bobbahbrown) [inline code/modules/mob/living/carbon/life.dm]:
```suggestion
	// the air you breath out should match your body temperature
```

File: code/modules/mob/living/carbon/life.dm [hunk 3]
```diff
@@ -302,7 +298,13 @@
 
 //Fourth and final link in a breath chain
 /mob/living/carbon/proc/handle_breath_temperature(datum/gas_mixture/breath)
-	return
+	// the air you breath out should match your body tempurature
+	if(bodytemperature > breath.temperature)
+		// raise the tempurature of our breath to just below body temp
```
Comment (bobbahbrown) [inline code/modules/mob/living/carbon/life.dm]:
```suggestion
		// raise the temperature of our breath to just below body temp
```

File: code/modules/surgery/organs/lungs.dm [hunk 4]
```diff
@@ -388,6 +388,14 @@
 			if(prob(20))
 				to_chat(H, "<span class='warning'>You feel [hot_message] in your [name]!</span>")
 
+	// the air you breath out should match your body tempurature
```
Comment (bobbahbrown) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
	// the air you breath out should match your body temperature
```

File: code/modules/surgery/organs/lungs.dm [hunk 5]
```diff
@@ -388,6 +388,14 @@
...
+	if(H.bodytemperature > breath.temperature)
+		// raise the tempurature of our breath to just below body temp
```
Comment (bobbahbrown) [inline code/modules/surgery/organs/lungs.dm]:
```suggestion
		// raise the temperature of our breath to just below body temp
```

**Review by bobbah — COMMENTED** [review]
consistency is next to godliness

File: code/modules/surgery/organs/lungs.dm (hunk 5 above)
Comment (nightred) [reply code/modules/surgery/organs/lungs.dm]: I have fixed all the changes, and reworded a bit.

File: code/modules/mob/living/carbon/life.dm (hunk 2 above)
Comment (cacogen) [reply code/modules/mob/living/carbon/life.dm]: breathe

File: code/modules/mob/living/carbon/life.dm [hunk 6]
```diff
@@ -302,7 +298,13 @@
...
 //Fourth and final link in a breath chain
 /mob/living/carbon/proc/handle_breath_temperature(datum/gas_mixture/breath)
-	return
+	// The air you breath out should about match your body temperature
+	if(bodytemperature > breath.temperature)
+		// Raise the temperature of our breath to just below body temp
+		breath.temperature = bodytemperature - BREATH_TEMPERATURE_OFFSET
+	else
+		// The air is hotter than out body so we need to lower the air temp
+		breath.temperature = bodytemperature + BREATH_TEMPERATURE_OFFSET
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/life.dm]: Is the offset necessary? Pretty sure air you breathe in quickly heats to exactly your internal temperature. Plus this creates edge cases where you could potentially breathe out a temperature that is a few degrees higher/lower than both the air and your body.

File: code/modules/surgery/organs/lungs.dm [hunk 7]
```diff
@@ -388,6 +388,14 @@
 			if(prob(20))
 				to_chat(H, "<span class='warning'>You feel [hot_message] in your [name]!</span>")
 
+	// The air you breath out should about match your body temperature
+	if(H.bodytemperature > breath.temperature)
+		// Raise the temperature of our breath to just below body temp
+		breath.temperature = H.bodytemperature - BREATH_TEMPERATURE_OFFSET
+	else
+		// The air is hotter than out body so we need to lower the air temp
+		breath.temperature = H.bodytemperature + BREATH_TEMPERATURE_OFFSET
```
Comment (Cyberboss) [inline code/modules/surgery/organs/lungs.dm]: Same comments

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/life.dm (hunk 6 above)
Comment (nightred) [reply code/modules/mob/living/carbon/life.dm]: This was done for a bit more real style as your breath is a touch lower.
I have no issue removing that.

File: code/modules/surgery/organs/lungs.dm (hunk 7 above)
Comment (nightred) [reply code/modules/surgery/organs/lungs.dm]: Both changes have been done.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**